tests/test_group.py

```diff
@@ -0,0 +1,63 @@
+import greenplumpython as gp
+from tests import db
+
+
+def test_group_agg(db: gp.Database):
+    rows = [(i, i % 2 == 0) for i in range(10)]
+    numbers = gp.values(rows, db=db, column_names=["val", "is_even"])
+    count = gp.aggregate("count", db=db)
+
+    results = list(numbers.group_by("is_even").aggregate(count, numbers["val"]).to_table().fetch())
+    assert len(results) == 2
+    for row in results:
+        assert ("is_even" in row) and (row["is_even"] is not None) and (row["count"] == 5)
+
+
+def test_group_by_multi_columns(db: gp.Database):
+    rows = [(i, i % 2 == 0, i % 3 == 0) for i in range(6)]  # 0, 1, 2, 3, 4, 5
+    numbers = gp.values(rows, db=db, column_names=["val", "is_even", "is_multiple_of_3"])
+    count = gp.aggregate("count", db=db)
+
+    results = list(
+        numbers.group_by("is_even", "is_multiple_of_3")
+        .aggregate(count, numbers["val"])
+        .to_table()
+        .fetch()
+    )
+    assert len(results) == 2 * 2
+    for row in results:
+        assert (
+            ("is_even" in row)
+            and (row["is_even"] is not None)
+            and ("is_multiple_of_3" in row)
+            and (row["is_multiple_of_3"] is not None)
+        )
+        assert (
+            (row["is_even"] and row["is_multiple_of_3"] and row["count"] == 1)  # 0
+            or (row["is_even"] and not row["is_multiple_of_3"] and row["count"] == 2)  # 2, 4
+            or (not row["is_even"] and row["is_multiple_of_3"] and row["count"] == 1)  # 3
+            or (not row["is_even"] and not row["is_multiple_of_3"] and row["count"] == 2)  # 1, 5
+        )
```

Comment:
I think we can just compare `row["is_even"]` and `row["is_multiple_of_3"]` with boolean values to make sure they are not `None`?
```suggestion
        assert (
            (row["is_even"] == True and row["is_multiple_of_3"] == True and row["count"] == 1)  # 0
            or (row["is_even"] == True and row["is_multiple_of_3"] == False and row["count"] == 2)  # 2, 4
            or (row["is_even"] == False and row["is_multiple_of_3"] == True and row["count"] == 1)  # 3
            or (row["is_even"] == False and row["is_multiple_of_3"] == False and row["count"] == 2)  # 1, 5
        )
```